src/mlpack/methods/ann/layer/lstm.hpp

```diff
@@ -171,6 +171,9 @@ class LSTM
   //! Get the number of output units.
   size_t OutSize() const { return outSize; }
 
+ //! Get size of weights.
+  size_t WeightSize() const { return 4 * outSize * inSize + 7 * outSize +
```

Comment:
I think it would be nice to change the calls to `weights.set_size()` in `lstm_impl.hpp` to `weights.set_size(WeightSize(), 1)`.  That way, the code that controls the size of the parameters is only in one place and it can't get out of sync. :+1:  (This is true for the other layers, too. :))